common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java

```diff
@@ -87,15 +92,16 @@ public static void submitSilent(@NonNull final BaseCommand command) {
                 "Beginning execution of silent command."
         );
 
-        if (sExecutingCommands.contains(command)) {
-            new Handler(Looper.getMainLooper()).post(new Runnable() {
-                @Override
-                public void run() {
-                    command.getCallback().onError(new ClientException(ClientException.DUPLICATE_COMMAND, "The same command was already received and is being processed."));
-                }
-            });
-        } else if (command.isEligibleForCaching()) {
-            sExecutingCommands.add(command);
+        final Handler handler = new Handler(Looper.getMainLooper());
+        ResultFuture<CommandResult> future = sExecutingCommandMap.get(command);
+
+        if (null != future) {
+            future.whenComplete(getCommandResultConsumer(command, handler));
+            return;
+        } else {
+            final ResultFuture<CommandResult> resultFuture = new ResultFuture<>();
+            resultFuture.whenComplete(getCommandResultConsumer(command, handler));
+            sExecutingCommandMap.put(command, resultFuture);
         }
 
         sSilentExecutor.execute(new Runnable() {
```

Comment:
Hm.  OK, so we *might* want to pass in this executor to whenComplete.  As you may notice, in the event of a complete request, we're going to run the call back on whatever the current thread is running on.  If that's OK, it's more straightforward.  But if we want our callback to run on the silentExecutor thread in the event that the result is already completed by the time we attempt to run it, we probably want to pass this executor into whenComplete to use in the event we hit that case.

Reply:
Correct me if I'm wrong here.... I think we want to leave as-is, as calling back on the `silentExecutor` would be a threading behavioral change

Reply:
returnCommandResult is called by the runnable that was submitted previously, I thought.  Unless it has a way to transfer back to the calling thread, which I don't see, we currently call back from that executor.

Reply:
It posts back on the `Handler` created at ~L94 see:
https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/iambmelt/command-sub-fix-905-redux/common/src/main/java/com/microsoft/identity/common/internal/controllers/CommandDispatcher.java#L230

Reply:
Wild.  OK, then it doesn't matter. Good - it's simpler like this.

Reply:
gg